paddle/phi/kernels/onednn/matmul_kernel.cc

```diff
@@ -0,0 +1,450 @@
+      Execute();
+      return *(mul_);
+    }
+
+    auto src_desc =
+        CreateMemDescriptor<XT>(&x_matrix, funcs::OneDNNMemoryFormat::nc);
+    x_input_ = CreateMemory<XT>(src_desc, &x_matrix);
+
+    if (is_int8_) {
+      const auto trans_y = TransposeInputY(&y_matrix);
+      auto scale_y =
+          ctx.HasDnnAttr("scale_y")
+              ? PADDLE_GET_CONST(std::vector<float>, ctx.GetDnnAttr("scale_y"))
+              : std::vector<float>();
+      y_input_ = QuantInputY(trans_y, scale_y);
+    } else {
+      y_input_ = TransposeInputY(&y_matrix);
+    }
+
+    auto dst_desc =
+        CreateMemDescriptor<OT>(output, funcs::OneDNNMemoryFormat::any);
+
+    mul_ = CreateMulPrimitive(*x_input_, *y_input_, dst_desc, output, ctx);
+    Execute();
+    return *(mul_);
+  }
+
+ private:
+  memory ReorderWithScale(const memory::desc &src_desc,
+                          const memory::desc &dst_desc,
+                          void *src_data,
+                          const std::vector<float> &scale) {
+    auto mask = scale.size() > 1 ? 1 : 0;
+    dnnl::primitive_attr attr;
+    attr.set_output_scales(mask, scale);
+
+    auto src_mem = memory(src_desc, engine_, src_data);
+    auto dst_mem = memory(dst_desc, engine_);
+
+    auto reorder_pd = dnnl::reorder::primitive_desc(src_mem, dst_mem, attr);
+
+    auto reorder = dnnl::reorder(reorder_pd);
+
+    auto &astream = OneDNNContext::tls().get_stream();
+    {
+      paddle::platform::RecordEvent record_reorder(
+          "int_reorder",
+          paddle::platform::TracerEventType::UserDefined,
+          2,
+          paddle::platform::EventRole::kUniqueOp);
+      reorder.execute(astream, src_mem, dst_mem);
+      astream.wait();
+    }
+
+    return dst_mem;
+  }
+
+  memory QuantInputY(memory input_y, const std::vector<float> &scale_y) {
+    const auto &dims = input_y.get_desc().data.dims;
+    auto ndims = input_y.get_desc().data.ndims;
+    auto y_dims = std::vector<int64_t>(dims, dims + ndims);
+
+    auto user_y_desc =
+        CreateMemDescriptor<YT>(y_dims, funcs::OneDNNMemoryFormat::oi);
+    auto y_desc =
+        CreateMemDescriptor<int8_t>(y_dims, funcs::OneDNNMemoryFormat::oi);
+
+    return ReorderWithScale(
+        user_y_desc, y_desc, input_y.get_data_handle(), scale_y);
+  }
+
+  dnnl::primitive_attr CreateMulAttr(const OneDNNContext &ctx,
+                                     bool force_fp32_output) {
+    dnnl::primitive_attr mul_attr;
+
+    auto scale_y_data =
+        ctx.HasDnnAttr("scale_y")
+            ? PADDLE_GET_CONST(std::vector<float>, ctx.GetDnnAttr("scale_y"))
+            : std::vector<float>();
+    auto scale_x_data = ctx.HasDnnAttr("scale_x")
+                            ? PADDLE_GET_CONST(float, ctx.GetDnnAttr("scale_x"))
+                            : 1.0f;
+    auto scale_out_data =
+        force_fp32_output ? 1.0f
+        : ctx.HasDnnAttr("scale_out")
+            ? PADDLE_GET_CONST(float, ctx.GetDnnAttr("scale_out"))
+            : 1.0f;
```

Comment:
```suggestion
  auto scale_out = ctx.HasDnnAttr("scale_out")
                       ? PADDLE_GET_CONST(float, ctx.GetDnnAttr("scale_out"))
                       : 1.0f;
  auto scale_out_data = force_fp32_output ? 1.0f : scale_out;
```
I think it's easier to read with two variables